taichi/transforms/constant_fold.cpp

```diff
@@ -70,30 +46,35 @@ class ConstantFold : public BasicStmtVisitor {
       current_ast_builder().insert(std::move(oper));
       current_ast_builder().insert(std::move(ret));
     };
-    auto ker = std::make_unique<Kernel>(get_current_program(), func, kernel_name);
+    auto ker =
+        std::make_unique<Kernel>(get_current_program(), func, kernel_name);
```

Comment:
We've probably mentioned this in the chat -- Is it preferable to always execute this kernel on CPU? That way it eschews the potential launching overhead on all GPUs?

Reply:
That's a good point. I thought about that as well - the issue is sometimes on CPUs and GPUs you will have different results, especially for float point numbers. E.g. `sin` on CUDA may have lower precision and a different result as `sin` on x64. I guess ultimately we will have to provide both options.